src/pinnable_mapped_file.cpp

```diff
@@ -122,8 +173,10 @@ pinnable_mapped_file::pinnable_mapped_file(const bfs::path& dir, bool writable,
 
          if(mode == locked) {
 #ifndef _WIN32
-            if(mlock(_mapped_region.get_address(), _mapped_region.get_size()))
-               BOOST_THROW_EXCEPTION(std::runtime_error("Failed to mlock database \"" + _database_name + "\""));
+            if(mlock(_mapped_region.get_address(), _mapped_region.get_size())) {
+	       std::cerr << "Failed to mlock database \"" <<  _database_name << "\"" << std::endl;
+               BOOST_THROW_EXCEPTION(std::system_error(make_error_code(db_error_code::no_mlock)));
```

Comment:
Instead of cerr create a string with this info and pass it into the system_error constructor